docs/guides/upgrading.rst

```diff
@@ -12,6 +12,42 @@ See the administrator guides for :doc:`how to upgrade a live site </admin/upgrad
 From 1.x to 2.0
 ===============
 
+JS and CSS views have been moved out of the js/ and css/ directories
+--------------------------------------------------------------------
+
+They also have been given .js and .css extensions respectively if they didn't
+already have them. For example:
+
+================= =============
+Old view          New view
+================= =============
+``js/view``       ``view.js``
+``css/view``      ``view.css``
+``js/other.js``   ``other.js``
+``css/other.css`` ``other.css``
+``js/img.png``    ``img.png``
+================= =============
+
+The main benefit this brings is being able to co-locate related assets.
+So a template (``view.php``) can have its CSS/JS dependencies right next to it
+(``view.css``, ``view.js``).
+
+Compare to the current approach which is like so for the case of ``.elgg-module``:
+
+ * Template: ``page/components/module.php``
+ * CSS: ``css/elements/modules.php``
+ * JS: None in this case, but how would you know with any confidence?
+
+In order to make this change, we had to re-configure view location storage such
+that full paths are stored, not just paths to view directories. 
+So ``elgg_get_view_location`` and ``elgg_set_view_location`` return and expect full
```

Comment:
elgg_set_view_location() would be easy to keep working, and elgg_get_view_location() could still work in many cases, though best to get rid of it.